### PR TITLE
Add an option to run MOM6 with ocean biogeochemistry (BLING model)

### LINKED ISSUE #17: Add an option to run MOM6 with ocean biogeochemistry (BLING model)
## Description

Add an option to build and run MOM6 with ocean biogeochemistry (BLING model). This is a preparation step for ocean color data assimilation development and activities.

## Solution

- in soca-bundle CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to git clone ocean_bgc repo recursively.

- in ocean_bgc CMakelist, export a list of ocean_bgc_src_files.

- in mom6 CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to import ocean_bgc_src_files and build them into mom6 libraries and executable.

- in soca, include BLING tracers in the MOM.res.nc restart file, use MOM_override to activate generic tracer packages, add an input_bgc.nml to set up ocean_bgc options (e.g. use BLING or other models), then add an optional new ctest forecast_mom6_bgc.test (if ENABLE_OCEAN_BGC = ON).

### LINKED ISSUE #394: Add an option to run MOM6 with ocean biogeochemistry (BLING model)
## Description

Add an option to build and run MOM6 with ocean biogeochemistry (BLING model). This is a preparation step for ocean color data assimilation development and activities.

## Solution

- in soca-bundle CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to git clone ocean_bgc repo recursively.

- in ocean_bgc CMakelist, export a list of ocean_bgc_src_files.

- in mom6 CMakelist, add an option (if ENABLE_OCEAN_BGC = ON) to import ocean_bgc_src_files and build them into mom6 libraries and executable.

- in soca, include BLING tracers in the MOM.res.nc restart file, use MOM_override to activate generic tracer packages, add an input_bgc.nml to set up ocean_bgc options (e.g. use BLING or other models), then add an optional new ctest forecast_mom6_bgc.test (if ENABLE_OCEAN_BGC = ON).

### PR DESCRIPTION
**Pull request by liuxiao37k**
## Description

This is one of the two PRs (in mom6 and soca, respectively) in order to add an option to build and run MOM6 with ocean biogeochemistry (BLING model). To do so, the following steps are needed:

- in ocean_bgc CMakelist, export a list of ocean_bgc_src_files (already merged to the ocean_bgc repo).

- add ocean_bgc as a submodule (pointer to the repo) in mom6 under /pkg (this PR)

- in mom6 CMakelist (this PR), add an option (if ENABLE_OCEAN_BGC = ON) to import ocean_bgc_src_files and build them into mom6 libraries and executable.

- in soca, include BLING tracers in the MOM.res.nc restart file, use MOM_override to activate generic tracer packages, append input.nml to set up ocean_bgc options (e.g. use BLING or other models), then add an optional new ctest forecast_mom6_bgc.test (if ENABLE_OCEAN_BGC = ON) to run MOM6 with BLING.

### Issue(s) addressed

- fixes #17 
- fixes JCSDA/soca/issues/394

## Testing

The two PRs were tested together on random Hera computing nodes:
ecbuild --build=release -DMPIEXEC_EXECUTABLE=`which srun` -DMPIEXEC_NUMPROC_FLAG="-n" -DBUILD_ECKIT=ON -DBUILD_CRTM=OFF -DENABLE_OCEAN_BGC=ON (soca_bundle_dir)

All 48 ctests (47 if -DENABLE_OCEAN_BGC=OFF) passed.

## Dependencies

- waiting on JCSDA/soca/pull/395

### CODE REVIEW

**Comment by aerorahul** [comment]
@liuxiao37k 
You need to add:
```
[submodule "pkg/ocean_BGC"]
	path = pkg/ocean_BGC
	url = https://github.com/NOAA-EMC/ocean_BGC.git
        branch = noaa-emc
```

to `.gitmodules`.

Cloning does not mean we have to compile with this option.  The changes in `CMakeLists.txt` look good.

File: CMakeLists.txt [hunk 1]
```diff
@@ -71,6 +71,10 @@ include_directories( ${FMS_INCLUDE_DIRS} )
 ecbuild_use_package( PROJECT gsw REQUIRED )
 include_directories( ${GSW_INCLUDE_DIRS} )
 
+# ocean_bgc
```
Comment (aerorahul) [inline CMakeLists.txt:74]: please add
```
ecbuild_add_option( FEATURE ENABLE_OCEAN_BGC
                    DEFAULT OFF
                    DESCRIPTION "download, build, and use ocean bgc modules in MOM6" )
```

**Comment by liuxiao37k** [comment]
> @liuxiao37k
> You need to add:
> 
> ```
> [submodule "pkg/ocean_BGC"]
> 	path = pkg/ocean_BGC
> 	url = https://github.com/NOAA-EMC/ocean_BGC.git
>         branch = noaa-emc
> ```
> 
> to `.gitmodules`.
> 
> Cloning does not mean we have to compile with this option. The changes in `CMakeLists.txt` look good.

Great, thanks! It turned out that this info can be automatically populated into .gitmodules after ocean_BGC is added as a submodule (pointer to the repo) under /pkg.

**Comment by liuxiao37k** [comment]
@aerorahul Made the changes as suggested. Thank you!

**Review by aerorahul — APPROVED** [review]
Thank you @liuxiao37k.
Changes look good.

**Comment by liuxiao37k** [comment]
> Thank you @liuxiao37k.
> Changes look good.

Thanks for the email. Sorry I forgot to push the changes in ocean_bgc/CMakeLists before adding the submodule to mom6. All updated now. Can you please check again? Thanks! https://github.com/NOAA-EMC/ocean_BGC/tree/ee0e7e436b1d711560a66b25aa9719154d911d90

**Comment by aerorahul** [comment]
> > Thank you @liuxiao37k.
> > Changes look good.
> 
> Thanks for the email. Sorry I forgot to push the changes in ocean_bgc/CMakeLists before adding the submodule to mom6. All updated now. Can you please check again? Thanks! https://github.com/NOAA-EMC/ocean_BGC/tree/ee0e7e436b1d711560a66b25aa9719154d911d90

@liuxiao37k 
ocean_bgc does not know what `pkg` is in `CMakeLists.txt`.  It is better to leave it as `${CMAKE_CURRENT_SOURCE_DIR}` instead of `pkg/ocean_bgc`

**Comment by liuxiao37k** [comment]
> > > Thank you @liuxiao37k.
> > > Changes look good.
> > 
> > 
> > Thanks for the email. Sorry I forgot to push the changes in ocean_bgc/CMakeLists before adding the submodule to mom6. All updated now. Can you please check again? Thanks! https://github.com/NOAA-EMC/ocean_BGC/tree/ee0e7e436b1d711560a66b25aa9719154d911d90
> 
> @liuxiao37k
> ocean_bgc does not know what `pkg` is in `CMakeLists.txt`. It is better to leave it as `${CMAKE_CURRENT_SOURCE_DIR}` instead of `pkg/ocean_bgc`

@aerorahul updated ocean_bgc CMakelist and submodule version in mom6! thank you!

**Review by aerorahul — APPROVED** [review]
looks good.

**Pull request merged**